lib/grizzly/zwave/security.ex

```diff
@@ -74,6 +74,7 @@ defmodule Grizzly.ZWave.Security do
   def failed_type_from_byte(0x08), do: :get
   def failed_type_from_byte(0x09), do: :verify
   def failed_type_from_byte(0x0A), do: :report
+  def failed_type_from_byte(_byte), do: :unk
```

Comment:
According to the spec this should never be true as all the bytes the spec says are accounted for.

Can you share what byte you received that prompted you to add this line?

Reply:
I saw 
@spec failed_type_from_byte(byte()) :: key_exchange_fail_type() | :unk
and presumed the line I added was missing.

Reply:
I don't understand "all the bytes the spec says are accounted for" given <<seq_number, status_byte, 0x00, new_node_id, granted_keys_byte, kex_fail_type_byte, dsk_binary::binary>>

Reply:
Okay I see, yeah the `:unk` should probably be removed in the typespec.

I meant that the Z-Wave spec says those are the only bytes for the fail type and those bytes only map to those atoms. So, if a device fails due to some other key failure, then it is not in line with the Z-Wave specification. Which we have learned that is possible, but I rather assume that a device not abiding by the Z-Wave spec is impossible until proven otherwise.

Reply:
Fixed

Reply:
But how is this particular case different from the many places where we do return a DecoderError if a byte does not conform to the ZWave specs?

Reply:
That's a good question. I think over time we will probably coverage the differences. I think I want to push more and more to decode error and just pass that to the caller so they can do whatever they want. I rather be in its own PR.